src/api/mixPanel.ts

```diff
@@ -0,0 +1,92 @@
+import { Agent } from "..";
+
+export interface AnalyticsOptions {
+    mixPanelKey: string;
+    agent: Agent;
+    chatId?: string;
+    isEnabled?: boolean;
+    distinct_id?: string;
+}
+
+class AnalyticsProvider {
+    private static instance: AnalyticsProvider;
+    private mixPanelKey: string;
+    private distinct_id?: string;
+    private isEnabled: boolean = true;
+    private chatId?: string;
+    private agentId: string;
+    private owner_id: string;
+
+    private constructor(config: AnalyticsOptions) {
+        this.mixPanelKey = config.mixPanelKey || 'testKey';
+        this.distinct_id = config.distinct_id || this.getUUID();
+        this.isEnabled = config.isEnabled || true;
+        this.chatId = config.chatId;
+        this.agentId = config.agent.id;
+        this.owner_id = config.agent.owner_id ?? ''
+    }
+
+    static getInstance(config?: AnalyticsOptions) {
+        if (!AnalyticsProvider.instance) {
+            if(!config) {
+                throw Error('Trying to use Analytics before initiaalize and without config')
+            }
+            AnalyticsProvider.instance = new AnalyticsProvider(config);
+        }
+        return AnalyticsProvider.instance;
+    }
+
+    setChatId(id: string) {
+        this.chatId = id;
+    }
+
+    getRandom() {
+        return Math.random().toString(16).slice(2);
+    }
+
+    getUUID() {
+        const trackingId = localStorage.getItem('tracking_id') ?? this.getRandom();
```

Comment:
i dont think its relevant anymore
is it? lets remove if not

Reply:
It's revelant, since Studio don't have code to set externalId, and in embeded UI we get `distinct_id` in this way

Reply:
no true
studio send external_id as well
verify me in `packages/apps/studio/src/pages/agents/share/AgentShare.tsx`
that why this always hsould be external_id gets from the outside

Reply:
In packages/apps/studio/src/pages/agents/share/AgentShare.tsx there is no usage external_id or tracking_id. They are using MixPanel SDK, and external_id request param is probably set inside it (            `mixpanel.people.set(mixpanelProps) `in file 
packages/apps/studio/src/services/analytics/analytics.tsx)